snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -111,64 +144,59 @@ def get_build_snaps(self) -> Set[str]:
-                "else",
-                "    echo 'Matter repository already exists, skip clone';",
-                "fi",
-                "cd matter;",
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h",
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h",
             ]
         )
 
-        # Checkout submodules for Linux platform
-        commands.extend(["scripts/checkout_submodules.py --shallow --platform linux"])
-
-        """
-        The project writes its data to /tmp which isn't persisted.
+        # Capture initial environment variables
+        commands.extend(["initial_environment=$(printenv | tr ' ' '\n' | sort)"])
 
-        Setting TMPDIR env var when running the app isn't sufficient as
-         chip_[config,counter,factory,kvs].ini still get written under /tmp.
-         The chip-tool currently has no way of overriding the default paths to
-         storage and security config files.
+        # Bootstrapping script for building Matter SDK with minimal "build" requirements
+        # and setting up the environment.
+        commands.extend(
+            ["set +u && source scripts/setup/bootstrap.sh --platform build && set -u"]
+        )
+        commands.extend(["echo 'Built Matter SDK'"])
 
-        Snap does not allow bind mounting a persistent directory on /tmp,
-         so we need to replace it in the source with another path, e.g. /mnt.
-         See the top-level layout definition which bind mounts a persisted
-         directory within the confined snap space on /mnt.
-        """
+        # Capture updated environment variables after the bootstrapping
+        commands.extend(["updated_environment=$(printenv | tr ' ' '\n' | sort)"])
 
-        # Replace storage paths
+        # Compare and output environment variable differences to matter_sdk_env env file
         commands.extend(
             [
-                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h",
-                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h",
+                f"environment_differences=$(comm -3 <(echo $initial_environment | tr ' ' '\n') \\",
+                f"<(echo $updated_environment | tr ' ' '\n') | awk '{{print $1}}')",
+                "for env in $environment_differences; do",
+                'echo "export $env" >> matter_sdk_env',
+                "done",
+                "echo 'Environment variables differences exported to matter_sdk_env file'",
             ]
         )
 
-        """
-        Bootstrapping script for building Matter SDK with minimal "build" requirements
-         and setting up the environment.
-        """
         commands.extend(
-            ["set +u && source setup/bootstrap.sh --platform build && set -u"]
+            [
+                'echo "export PATH=$PATH" >> matter_sdk_env',
+                "echo 'Environment variable PATH exported to matter_sdk_env file'",
+            ]
```

Comment:
This is a very complicated way of handling this. It appears that the relevant environment variables contain CIPD and PW in their names. That should make it easy to filter and write them to the file. What remains is ZAP_INSTALL_PATH (see previous comment) and the modified PATH.

Reply:
Is the following method less complicated?
```
commands.extend([
    'echo "export PATH=$PATH" >> matter_sdk_env',
    'env | grep "^PW_" | sed \'s/^/export /\' >> matter_sdk_env',
    'env | grep "^_PW" | sed \'s/^/export /\' >> matter_sdk_env',
    'echo "export VIRTUAL_ENV=$VIRTUAL_ENV" >> matter_sdk_env',
    'echo "export CIPD=$CIPD" >> matter_sdk_env',
    'echo "export ZAP_INSTALL_PATH=$ZAP_INSTALL_PATH" >> matter_sdk_env',
    "echo 'Specific environment variables differences exported to matter_sdk_env file'",
])
```

Reply:
It sure is.

For PW, it may be safe to filter any variable containing `PW_`.

Also, keep in mind that because of the `export` keywords, this isn't an env file, but a bash script. In this case, the bash script can simply use `set -a/set +a` to export all the local variables, instead of injecting `export`s to each line.

Reply:
ok, Updated.